sql/2021/resource-hints/preconnect_hosts.sql

```diff
@@ -0,0 +1,45 @@
+#standardSQL
+# Most popular hosts users preconnect to
+
+
+CREATE TEMPORARY FUNCTION getResourceHintsHrefs(payload STRING, hint STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var almanac = JSON.parse($._almanac);
+  return almanac['link-nodes'].nodes.filter(link => link.rel.toLowerCase() == hint).map(n => n.href);
+} catch (e) {  
+  return [];
+}
+''' ;
+
+SELECT
+  client,
+  host,
+  COUNT(0) AS freq,
+  COUNT(0) / SUM(COUNT(0)) OVER () AS pct
```

Comment:
Omitting the `PARTITION BY client` will add desktop and mobile results together.

Also a good idea to always include the total in the results when calculating percents.
```suggestion
  SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
```